src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,177 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 
-public class Range<T> implements Set<T> {
+public class Range<T> implements Set<T> , Comparable {
+
+    static Object[] elementData = new Object[10];
+    static int modCount = 0;
+
+    static Number left;
+    static Number right;
+    boolean isEmpty = true;
+
+    public static Range<Integer> of(Integer a, Integer b) {
+        Range<Integer> range = new Range<>();
+        range.clear();
+        left = a;
+        right = b;
+        for(Integer j = left.intValue(); j<=right.intValue(); j++){
+            range.add(j);
+        }
+        return range;
+
+    }
+
+    public static Range<Float> of(Float a, Float b) {
+        Range<Float> range = new Range<>();
+        range.clear();
+        left = a;
+        right = b;
+        for(Float j = left.floatValue(); j<=right.floatValue(); j=j+0.1f){
+            range.add(j);
+        }
+        return range;
+
+    }
+
+    public static Range<Character> of(char a, char d, Function<Character, Character> characterCharacterFunction) {
+        Range<Character> range  = new Range<>();
+        range.clear();
+        for(Character j = a; j<=d; j++){
+            range.add(j);
+        }
+        return range;
+    }
 
     public int size() {
-        return 0;
+        return modCount;
     }
 
     public boolean isEmpty() {
-        return false;
+        return this.isEmpty;
     }
 
     public boolean contains(Object o) {
-        return false;
+        return java.util.Arrays.asList(elementData).indexOf(o) >= 0;
     }
 
     public Iterator<T> iterator() {
-        return null;
+        Iterator iterator = new Iterator() {
+            int cursor = 0;
+            int lastRet = -1;
+            @Override
+            public boolean hasNext() {
+                return cursor != modCount;
+            }
+
+            @Override
+            public Object next() {
+                int i = cursor;
+                if (i >= modCount)
+                    throw new NoSuchElementException();
+                if (i >= elementData.length)
+                    throw new ConcurrentModificationException();
+                cursor = i + 1;
+                return (Object) elementData[lastRet = i];
+            }
+        };
+        return iterator;
     }
 
     public Object[] toArray() {
-        return new Object[0];
+        return elementData;
     }
 
     public <T1> T1[] toArray(T1[] a) {
-        return null;
+        T1[] array = (T1[]) new Object[a.length];
```

Comment:
You already got new array as a parameter (see T1[] a), just have to fill it with your elements.